integration-tests/utils/aws/cloudWatchGetLogs.ts

```diff
@@ -1,141 +1,116 @@
-import { INITIATE_DATA_REQUEST_LAMBDA_LOG_GROUP } from '../../constants/awsParameters'
 import { cloudWatchLogsClient } from './cloudWatchLogsClient'
 import {
-  DescribeLogStreamsCommandInput,
-  DescribeLogStreamsCommandOutput,
   DescribeLogStreamsCommand,
   LogStream,
-  FilterLogEventsCommandInput,
   FilterLogEventsCommand,
-  FilterLogEventsCommandOutput,
   FilteredLogEvent
 } from '@aws-sdk/client-cloudwatch-logs'
 
-export const getLogStreamPrefix = () => {
-  const dateFormat = new Intl.DateTimeFormat('en-GB')
-  const dateParts = dateFormat.format(new Date()).split('/')
-  return `${
-    dateParts[2] + '/' + dateParts[1] + '/' + dateParts[0] + '/' + '[$LATEST]'
-  }`
-}
-
-const getLatestLogStreamName = async (): Promise<string> => {
-  const describeLogStreamsParams: DescribeLogStreamsCommandInput = {
-    logGroupName: INITIATE_DATA_REQUEST_LAMBDA_LOG_GROUP,
-    orderBy: 'LastEventTime',
-    descending: true,
-    limit: 1
-  }
-
-  const describeLogStreamsCommand = new DescribeLogStreamsCommand(
-    describeLogStreamsParams
+export const getCloudWatchLogEventsGroupByMessagePattern = async (
+  logGroupName: string,
+  eventMessagePatterns: string[],
+  maxAttempts = 30
+) => {
+  const event = await waitForEventWithPatterns(
+    logGroupName,
+    eventMessagePatterns,
+    maxAttempts
   )
 
-  await pause(500)
+  if (!event) return []
 
-  const describeLogStreamsResponse: DescribeLogStreamsCommandOutput =
-    await cloudWatchLogsClient.send(describeLogStreamsCommand)
+  const requestId = extractRequestIdFromEventMessage(event.message as string)
```

Comment:
This might not be a string.

You could tweak the check on line 20 to make sure it's what your expecting.  Something like: 
`  if (!event || !event.message || !event.logStreamName) return []
`

Might be worth moving into a separate function.

Reply:
Done, now `if (!event || !event.message) return []` since `logStreamName` is always there (have removed the unnecessary cast)